pennylane_lightning/lightning_gpu/lightning_gpu.py

```diff
@@ -0,0 +1,754 @@
+
+        def _apply_basis_state(self, state, wires):
+            """Initialize the state vector in a specified computational basis state on GPU directly.
+             Args:
+                state (array[int]): computational basis state (on host) of shape ``(wires,)``
+                    consisting of 0s and 1s.
+                wires (Wires): wires that the provided computational state should be initialized on
+            Note: This function does not support broadcasted inputs yet.
+            """
+            # translate to wire labels used by device
+            device_wires = self.map_wires(wires)
+
+            # length of basis state parameter
+            n_basis_state = len(state)
+            state = state.tolist() if hasattr(state, "tolist") else state
+            if not set(state).issubset({0, 1}):
+                raise ValueError("BasisState parameter must consist of 0 or 1 integers.")
+
+            if n_basis_state != len(device_wires):
+                raise ValueError("BasisState parameter and wires must be of equal length.")
+
+            # get computational basis state number
+            basis_states = 2 ** (self.num_wires - 1 - np.array(device_wires))
+            basis_states = qml.math.convert_like(basis_states, state)
+            num = int(qml.math.dot(state, basis_states))
+
+            self._create_basis_state(num)
+
+        # pylint: disable=missing-function-docstring
+        def apply_lightning(self, operations):
+            """Apply a list of operations to the state tensor.
+
+            Args:
+                operations (list[~pennylane.operation.Operation]): operations to apply
+                dtype (type): Type of numpy ``complex`` to be used. Can be important
+                to specify for large systems for memory allocation purposes.
+
+            Returns:
+                array[complex]: the output state tensor
+            """
+            # Skip over identity operations instead of performing
+            # matrix multiplication with the identity.
+            skipped_ops = ["Identity"]
+            invert_param = False
+
+            for ops in operations:
+                if str(ops.name) in skipped_ops:
+                    continue
+                name = ops.name
+                if isinstance(ops, Adjoint):
+                    name = ops.base.name
+                    invert_param = True
+                method = getattr(self._gpu_state, name, None)
+                wires = self.wires.indices(ops.wires)
+
+                if method is None:
+                    # Inverse can be set to False since qml.matrix(ops) is already in inverted form
+                    try:
+                        mat = qml.matrix(ops)
+                    except AttributeError:  # pragma: no cover
+                        # To support older versions of PL
+                        mat = ops.matrix
+
+                    if len(mat) == 0:
+                        raise ValueError("Unsupported operation")
```

Comment:
We can likely add a quick test for this situation to get coverage

Reply:
Added!